pdns/backends/gsql/gsqlbackend.cc

```diff
@@ -1501,6 +1506,105 @@ string GSQLBackend::directBackendCmd(const string &query)
  }
 }
 
+bool GSQLBackend::searchRecords(const string &pattern, int maxResults, vector<DNSResourceRecord>& result)
+{
+  try {
+    string escaped_pattern = boost::replace_all_copy(pattern,"\\","\\\\");
+    boost::replace_all(escaped_pattern,"_","\\_");
+    boost::replace_all(escaped_pattern,"%","\\%");
+    boost::replace_all(escaped_pattern,"*","%");
+    boost::replace_all(escaped_pattern,"?","_");
+
+    d_SearchRecordsQuery_stmt->
+      bind("value", escaped_pattern)->
+      bind("value2", escaped_pattern)->
+      bind("limit", maxResults)->
+      execute();
+
+    while(d_SearchRecordsQuery_stmt->hasNextRow())
+    {
+      SSqlStatement::row_t row;
+      DNSResourceRecord r;
+      d_SearchRecordsQuery_stmt->nextRow(row);
+
+      if (row[1].empty())
+        r.ttl = ::arg().asNum( "default-ttl" );
+      else
+        r.ttl=atol(row[1].c_str());
+      if(!d_qname.empty())
+        r.qname=d_qname;
+      else
+        r.qname=row[6];
+      r.qtype=row[3];
+
+      if (r.qtype==QType::MX || r.qtype==QType::SRV)
+        r.content=row[2]+" "+row[0];
+      else
+        r.content=row[0];
+
+      r.last_modified=0;
+
+      if(d_dnssecQueries)
+        r.auth = !row[7].empty() && row[7][0]=='1';
+      else
+        r.auth = 1;
+
+      r.disabled = !row[5].empty() && row[5][0]=='1';
+
+      r.domain_id=atoi(row[4].c_str());
+      result.push_back(r);
+    }
+
+    d_SearchRecordsQuery_stmt->reset();
+
+    return true;
+  }
+  catch (SSqlException &e) {
+    throw PDNSException("GSQLBackend unable to execute query: "+e.txtReason());
+  }
+
+  return false;
+}
+
+bool GSQLBackend::searchComments(const string &pattern, int maxResults, vector<Comment>& result)
+{
+  Comment c;
+  try {
+    string escaped_pattern = boost::replace_all_copy(pattern,"\\","\\\\");
+    boost::replace_all(escaped_pattern,"_","\\_");
+    boost::replace_all(escaped_pattern,"%","\\%");
+    boost::replace_all(escaped_pattern,"*","%");
+    boost::replace_all(escaped_pattern,"?","_");
```

Comment:
move the replace_ stuff into a local helper function, for less code duplication?